ludwig/features/category_feature.py

```diff
@@ -137,13 +141,26 @@ def get_feature_meta(column, preprocessing_parameters, backend):
 
     @staticmethod
     def feature_data(column, metadata):
-        return column.map(
-            lambda x: (
-                metadata["str2idx"][x.strip()]
-                if x.strip() in metadata["str2idx"]
-                else metadata["str2idx"][UNKNOWN_SYMBOL]
+        def __replace_token_with_idx(value: Any, metadata: Dict[str, Any], unknown_symbol_idx: int) -> int:
```

Comment:
Rename `unknown_symbol_idx` to `fallback_symbol_idx` so that we can reserve the word "unknown" for the actual unknown symbol.